File: Programs in C/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

float countLetters(string text);
int countWords(string text);
float countSentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = countLetters(text);
    int words = countWords(text);
    int sentences = countSentences(text);

    // Compute the Coleman-Liau index

    float L = ((float) letters / (float) words) * 100;
    float S = ((float) sentences / (float) words) * 100;
    float index = (0.0588 * L - 0.296 * S - 15.8);
    if (index > 16)
    {
        printf(" Grade: 16+ \n");
        // Print the grade level
    }
    else
    {
        printf(" Grade: %f \n", round(index));

        // Print the grade level
    }
}

float countLetters(string text)
{

    int counter = 0;
    int letters = 0;

    // Return the number of letters in text

    for (int i = 0, t = strlen(text); i < t; i++)
    {
        if (text[i] != ' ')
        {
            counter++;
        }
        else
        {
            continue;
        }
    }
    letters = counter;

    return letters;
}

int countWords(string text)
{
    int words = 0;
    int whiteSpaces = 0;

    for (int i = 0, t = strlen(text); i < t; i++)
    {
        if (text[i] != ' ')
        {
            continue;
        }
        else
        {
            whiteSpaces++;
        }

        words = whiteSpaces + 1;
    }
    // Return the number of words in text

    return words;
}

float countSentences(string text)
{
    int sentences = 0;

    for (int i = 0, t = strlen(text); i < t; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
        else
        {
            continue;
        }
    }

    // Return the number of sentences in text
    return sentences;
}

File: Programs in Python/sentimental-readability/readability.py
from cs50 import get_string
while True:
    text = get_string("Text: ")
    length = len(text)

    # Get number of letters in text#
    letters = 0
    for i in range(length):
        if text[i] != ' ' and text[i] != "!" and text[i] != "." and text[i] != "?" and text[i] != "-" and text[i] != ";" and text[i] != "'" and text[i] != "," and text[i] != "(" and text[i] != ")":
            letters = letters + 1

    # Get number of word in text#
    words = 0
    for i in range(length):
        if text[i] == ' ':
            words = words + 1
    words = words + 1

    # Get number of sentences#
    sentences = 0
    for i in range(length):
        if text[i] == "!" or text[i] == "." or text[i] == "?":
            sentences = sentences + 1

    # Computee Coleman-Liau Index#
    L = float(letters / words) * 100
    S = float(sentences / words) * 100

    CLI = float(0.0588 * L - 0.296 * S - 15.8)

    # Print output#
    if CLI >= 16:
        print("Grade 16+")
    elif CLI < 1:
        print("Before Grade 1")
    else:
        print("Grade: ")
        print(round(CLI))

    break
